### ISSUE TITLE
valid workflow fails "An internal server error occurred"

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/cwlviewer#419]
## Description
https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl fails to visualize

## Expected Behavior
![Whole_Exome-Seq](https://user-images.githubusercontent.com/1330696/172583604-e6da9381-9811-4ff8-bece-d1fb21c06ef2.png)

## Actual Behavior
> Whoops - Something Went Wrong!
> Error: An internal server error occurred 

## Steps to Reproduce
Local rendering with `cwltool` works: 
`cwltool: cwltool --print-dot https://github.com/athenarc/whole-exome-seq/raw/main/workflows/Whole_Exome-Seq.cwl`

**Comment by kinow** [comment]
I used the `main` branch, where I found a problem with the entrypoint of the updated `Dockerfile`, but I think that hasn't been deployed to production yet.

Updating the entrypoint command, and using the URL `https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl `, it is imported successfully in my local environment.

![image](https://user-images.githubusercontent.com/304786/172598167-3982c350-8867-4f9f-9043-bbbda8b6835a.png)

![image](https://user-images.githubusercontent.com/304786/172598252-8fea76cd-4cbc-4eb8-938e-e22964855775.png)

![image](https://user-images.githubusercontent.com/304786/172598340-67a58b48-6bc8-4098-a3e9-e3935654ecbb.png)

Going to use the 1.4.2 tag now.

**Comment by kinow** [comment]
Exception in `1.4.3` Docker compose logs:

```bash
postgres_1_e5bdf7fd818a | 2022-06-08 11:00:17.832 UTC [57] LOG:  execute S_2: COMMIT
spring_1_2d9b71585be6 | 2022-06-08 11:00:18,021 ERROR [task-1] org.commonwl.view.cwl.CWLToolRunner: ERROR I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
spring_1_2d9b71585be6 | The error was: No plugin registered for (json-ld, <class 'rdflib.parser.Parser'>)
spring_1_2d9b71585be6 | 
spring_1_2d9b71585be6 | org.commonwl.view.cwl.CWLValidationException: ERROR I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
spring_1_2d9b71585be6 | The error was: No plugin registered for (json-ld, <class 'rdflib.parser.Parser'>)
spring_1_2d9b71585be6 | 
spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLTool.runCwltoolOnWorkflow(CWLTool.java:121)
spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLTool.getRDF(CWLTool.java:50)
spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLService.parseWorkflowWithCwltool(CWLService.java:293)
spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLToolRunner.createWorkflowFromQueued(CWLToolRunner.java:86)
spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLToolRunner$$FastClassBySpringCGLIB$$cd465805.invoke(<generated>)
spring_1_2d9b71585be6 | 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
spring_1_2d9b71585be6 | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
spring_1_2d9b71585be6 | 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
spring_1_2d9b71585be6 | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
spring_1_2d9b71585be6 | 	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115)
spring_1_2d9b71585be6 | 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
spring_1_2d9b71585be6 | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
spring_1_2d9b71585be6 | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
spring_1_2d9b71585be6 | 	at java.lang.Thread.run(Thread.java:748)
```

```bash
bash-4.4# cwltool --version
/usr/bin/cwltool 3.1.20220502060230

```

**Comment by kinow** [comment]
Installed `cwltool` with `pip install cwltool==3.1.20220502060230`. Then `cwltool --pack https://raw.githubusercontent.com/athenarc/whole-exome-seq/main/workflows/Whole_Exome-Seq.cwl`. Executed successfully.

**Comment by kinow** [comment]
Tried the other command executed in CWLViewer's `CWLTool`, `cwltool --non-strict --quiet --skip-schemas --print-rdf https://raw.githubusercontent.com/athenarc/whole-exome-seq/main/workflows/Whole_Exome-Seq.cwl`.

Also executed successfully :exploding_head:

**Comment by kinow** [comment]
Ah, I was using the raw URL. Reproduced the error in `cwltool`:

```bash
(venv) kinow@ranma:/tmp$ cwltool --non-strict --quiet --skip-schemas --print-rdf https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl
While fetching https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl, got content-type of 'text/html'. Expected one of ['text/plain', 'application/json', 'text/vnd.yaml', 'text/yaml', 'text/x-yaml', 'application/x-yaml', 'application/octet-stream'].
ERROR Tool definition failed validation:
https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl:28:66: mapping values are
                                                                                           not allowed here
(venv) kinow@ranma:/tmp$ cwltool --version
/tmp/venv/bin/cwltool 3.1.20220502060230

```

**Comment by kinow** [comment]
Huh, scary, now I realized the whole CWLViewer app is stuck. The server appears to be processing the steps in that workflow, even after it failed to execute `cwltool`. Either a loop, or something that is taking a really long time to execute.

**Comment by mr-c** [comment]
> Exception in `1.4.3` Docker compose logs:
> 
> ```shell
> postgres_1_e5bdf7fd818a | 2022-06-08 11:00:17.832 UTC [57] LOG:  execute S_2: COMMIT
> spring_1_2d9b71585be6 | 2022-06-08 11:00:18,021 ERROR [task-1] org.commonwl.view.cwl.CWLToolRunner: ERROR I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
> spring_1_2d9b71585be6 | The error was: No plugin registered for (json-ld, <class 'rdflib.parser.Parser'>)
> spring_1_2d9b71585be6 | 
> spring_1_2d9b71585be6 | org.commonwl.view.cwl.CWLValidationException: ERROR I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
> spring_1_2d9b71585be6 | The error was: No plugin registered for (json-ld, <class 'rdflib.parser.Parser'>)
> spring_1_2d9b71585be6 | 
> spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLTool.runCwltoolOnWorkflow(CWLTool.java:121)
> spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLTool.getRDF(CWLTool.java:50)
> spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLService.parseWorkflowWithCwltool(CWLService.java:293)
> spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLToolRunner.createWorkflowFromQueued(CWLToolRunner.java:86)
> spring_1_2d9b71585be6 | 	at org.commonwl.view.cwl.CWLToolRunner$$FastClassBySpringCGLIB$$cd465805.invoke(<generated>)
> spring_1_2d9b71585be6 | 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
> spring_1_2d9b71585be6 | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
> spring_1_2d9b71585be6 | 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
> spring_1_2d9b71585be6 | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
> spring_1_2d9b71585be6 | 	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115)
> spring_1_2d9b71585be6 | 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> spring_1_2d9b71585be6 | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> spring_1_2d9b71585be6 | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> spring_1_2d9b71585be6 | 	at java.lang.Thread.run(Thread.java:748)
> ```
> 
> ```shell
> bash-4.4# cwltool --version
> /usr/bin/cwltool 3.1.20220502060230
> ```

Thanks for reproducing this! Odd, that `cwltool` version matches the latest on quay.io, which is Python 3.10 based and shouldn't have that error: https://quay.io/repository/commonwl/cwltool/manifest/sha256:339d5687c91608e75016d65fcdb7dec8d5d24b4ac9e3547766a1db2faa5806d4

Ah, `quay.io/commonwl/cwltool` is not the container being used; it is https://github.com/common-workflow-language/cwlviewer/blob/d1646502a3a38eae0435bb2c4efe9662a4640154/Dockerfile#L1 → https://quay.io/repository/commonwl/cwlviewer/manifest/sha256:f44d3bb01a3f1b2818ac5b15dfaab4d6f4718dbc0b0191340f7a3ab825d8260d

> Ah, I was using the raw URL. Reproduced the error in `cwltool` on my local environment (Python 3.9, @mr-c ):
> 
> ```shell
> (venv) kinow@ranma:/tmp$ cwltool --non-strict --quiet --skip-schemas --print-rdf https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl
> While fetching https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl, got content-type of 'text/html'. Expected one of ['text/plain', 'application/json', 'text/vnd.yaml', 'text/yaml', 'text/x-yaml', 'application/x-yaml', 'application/octet-stream'].
> ERROR Tool definition failed validation:
> https://github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl:28:66: mapping values are
>                                                                                            not allowed here

Yes, that should fail. You asked `cwltool` to parse a webpage; which is why I was using the raw URL.

**Comment by mr-c** [comment]
Yep, the container's base image has an old Python version (3.6.9)

```
$ docker run  quay.io/commonwl/cwlviewer python3 -V
Python 3.6.9
```

**Comment by mr-c** [comment]
I'm rebuilding the container locally; it now pulls in a newer Python (3.9.7); I'm also making a new cwltool release that should fix the issue for Python 3.6

**Comment by mr-c** [comment]
Success! https://view.commonwl.org/workflows/github.com/athenarc/whole-exome-seq/blob/main/workflows/Whole_Exome-Seq.cwl

Thank you @kinow and @cure !
